library/Ivoz/Provider/Domain/Model/SurvivalDevice/SurvivalDevice.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Model\SurvivalDevice;
+
+use Assert\Assertion;
+
+/**
+ * SurvivalDevice
+ */
+class SurvivalDevice extends SurvivalDeviceAbstract implements SurvivalDeviceInterface
+{
+    use SurvivalDeviceTrait;
+
+    /**
+     * @codeCoverageIgnore
+     * @return array<string, mixed>
+     */
+    public function getChangeSet(): array
+    {
+        return parent::getChangeSet();
+    }
+
+    /**
+     * Get id
+     * @codeCoverageIgnore
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    protected function sanitizeValues(): void
+    {
+        $this->sanitizeProxyLogic();
+    }
+
+    protected function setUdpPort(int $udpPort): static
+    {
+        Assertion::between($udpPort, 1, 65535, "Invalid udp port number: $udpPort. Must be between 1 and 65535");
+        return parent::setUdpPort($udpPort);
+    }
+
+    protected function setTcpPort(int $tcpPort): static
+    {
+        Assertion::between($tcpPort, 1, 65535, "Invalid tcp port number: $tcpPort. Must be between 1 and 65535");
+        return parent::setTcpPort($tcpPort);
+    }
+
+    protected function setTlsPort(int $tlsPort): static
+    {
+        Assertion::between($tlsPort, 1, 65535, "Invalid tls port number: $tlsPort. Must be between 1 and 65535");
+        return parent::setTlsPort($tlsPort);
+    }
+
+    protected function setWssPort(int $wssPort): static
+    {
+        Assertion::between($wssPort, 1, 65535, "Invalid wss port number: $wssPort. Must be between 1 and 65535");
+        return parent::setWssPort($wssPort);
+    }
+
+    protected function setProxy(string $proxy): static
+    {
+        $proxy = trim($proxy);
+        if (empty($proxy)) {
+            throw new \DomainException('Proxy cannot be empty');
+        }
+
+        return parent::setProxy($proxy);
+    }
+
+    protected function sanitizeProxyLogic(): void
+    {
+        $outboundProxy = $this->getOutboundProxy();
+        if (is_null($outboundProxy)) {
+            return;
+        }
+
+        $emptyOutboundProxy = empty(
+            trim($outboundProxy)
+        );
+        if ($emptyOutboundProxy) {
+            throw new \DomainException('Outbound Proxy cannot be empty.');
+        }
+        if ($this->getProxy() && strpos($this->getProxy(), ':') !== false) {
+            throw new \DomainException('Proxy must not include a port when Outbound Proxy is set.', 70003);
+        }
```

Comment:
```suggestion
        $sipProxyIncludesPort = strpos($this->getProxy(), ':') !== false;

        if ($sipProxyIncludesPort) {
            throw new \DomainException('When Outbound Proxy is used, SIP Proxy must not include a port.', 70003);
        }                
```

1. We usually define variables to clarify our intentions. In this case "$sipProxy includes port" for we are checking whether proxy includes a port or not. And out condition know can read almost like in plain english "if sip proxy includes port..."
2. There's no need to check whether "getProxy()" returns truthy. It's always truthy if executions reaches this point, and implies forcing proxy to be non-empty out of place and with the wrong exception.
3. We should be using exactly the same exception message as in CarrierServer unless told otherwise.